client/src/components/games/GameDetails.tsx

```diff
@@ -1,44 +1,50 @@
-import { useContext, useEffect, useState } from "react"
+import React, { useContext, useEffect, useState } from "react"
 import { useParams } from "react-router-dom"
 import { GamesContext } from "../../App"
 import { getGame, joinGame } from "../../services/apiService"
 import { dateDisplay, timeDisplay } from "../../utilities/date-time-display"
 import NavBar from "../common/NavBar"
+import { Game } from '../../@types/game'
 
 function GameDetails() {
   const { games, setGames } = useContext(GamesContext)
-  const { gameId } = useParams()
+  const { gameId } = useParams();
 
-  const [game, setGame] = useState()
+  const [game, setGame] = useState<Game | null>(null)
   const [userName, setUserName] = useState('')
 
   useEffect(() => {
     (async () => {
-      const game = await getGame(gameId);
-      setGame(game);
+      if (gameId) {
+        const game = await getGame(gameId);
+        setGame(game);
+      }
     })();
   }, [gameId])
 
-  const handleChange = (event) => {
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     setUserName(event.target.value)
   }
 
-  const handleSubmit = async (event) => {
+  const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
     try {
-      console.log(userName);
-      const updatedGame = await joinGame(gameId, { userName });
+      if (!gameId) return;
+      
+      let updatedGame: Game;
+      updatedGame = await joinGame(gameId, { userName });

```

Comment:
Might be an alternative: `const updatedGame = await joinGame(gameId, { userName }) as Game;`